test/testenv/s3utils.go

```diff
@@ -64,27 +64,58 @@ func CheckPrefixExistsOnS3(prefix string) bool {
 	return false
 }
 
-// DownloadFromS3Bucket downloads license file from S3
-func DownloadFromS3Bucket() (string, error) {
-	dataBucket := testS3Bucket
+// DownloadLicenseFromS3Bucket downloads license file from S3
+func DownloadLicenseFromS3Bucket() (string, error) {
 	location := enterpriseLicenseLocation
 	item := "enterprise.lic"
-	file, err := os.Create(item)
+	filename, err := DownloadFileFromS3(item, location)
+	return filename, err
+}
+
+// S3Session Create session object for S3 bucket connection
+func S3Session() *session.Session {
+	sess, _ := session.NewSession(&aws.Config{Region: aws.String(s3Region)})
+	return sess
+}
+
+// DownloadFileFromS3 downloads file from S3
+func DownloadFileFromS3(filename string, filepath string) (string, error) {
+	dataBucket := testS3Bucket
+	file, err := os.Create(filename)
 	if err != nil {
-		logf.Log.Error(err, "Failed to create license file")
+		logf.Log.Error(err, "Failed to create file")
 	}
 	defer file.Close()
-	sess, _ := session.NewSession(&aws.Config{Region: aws.String(s3Region)})
+	sess := S3Session()
 	downloader := s3manager.NewDownloader(sess)
 	numBytes, err := downloader.Download(file,
 		&s3.GetObjectInput{
 			Bucket: aws.String(dataBucket),
-			Key:    aws.String(location + "/" + "enterprise.lic"),
+			Key:    aws.String(filepath + "/" + filename),
 		})
 	if err != nil {
-		logf.Log.Error(err, "Failed to download license file")
+		logf.Log.Error(err, "Failed to download file")
 	}
 
 	logf.Log.Info("Downloaded", "filename", file.Name(), "bytes", numBytes)
 	return file.Name(), err
 }
+
+// UploadFileToS3 upload file to S3
+func UploadFileToS3(filename string, filepath string, file *os.File) (string, error) {
+	dataBucket := testS3Bucket
```

Comment:
Would be a good idea to accept this as a parameter. Make the method more modular.

Reply:
Modified